### PR TITLE
Add template web.xml.j2 in authenticationendpoint

### PR DESCRIPTION
**Pull request by sumedhe**
## Purpose
Adding template web.xml.j2 in authenticationendpoint for following configurations

``` toml
[authenticationendpoint]
auth_rest_endpoint_url="https://localhost:9443/api/identity/auth/v1.1/"
registration_url="https://localhost:9443/accountrecoveryendpoint/register.do"
```

## Goals
Related:
* wso2-enterprise/asgardeo-product#912
* wso2-enterprise/asgardeo-app-extensions#100

### CODE REVIEW

File: features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2 [hunk 1]
```diff
@@ -0,0 +1,536 @@
+<?xml version="1.0" encoding="UTF-8"?><!--
+  ~ Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:2]:
```suggestion
  ~ Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

Comment (sumedhe) [reply features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2]: Fixed

File: features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+        <param-value>true</param-value>
+    </context-param-->
+    <!--context-param>
+        <param-name>EnableSelfSignUpEndpoint</param-name>
+        <param-value>true</param-value>
+    </context-param-->
+
+    {% if authenticationendpoint.registration_url is defined %}
+    <context-param>
+        <param-name>AccountRegisterEndpointURL</param-name>
```
Comment (mefarazath) [inline features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:72]: Is this endpoint used in on-prem product as well?

Comment (sumedhe) [reply features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:72]: Yes. That url is used in `Create Account` buttons of the login form. Recently it is modified to configure it from the web.xml (in #1496)

File: features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2 [hunk 3]
```diff
@@ -0,0 +1,536 @@
+<?xml version="1.0" encoding="UTF-8"?><!--
```
Comment (jeradrutnam) [inline features/org.wso2.identity.apps.authentication.portal.server.feature/resources/web.xml.j2:1]:
```suggestion
<?xml version="1.0" encoding="UTF-8"?>

<!--
```

**Review by senthalan — APPROVED** [review]

**Pull request merged**